### PR TITLE
test(e2e): add E2E tests for signout-modal ui component

### PR DESCRIPTION
**Pull request by sitek94**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Solves "ui-components/signout-modal" #51705

In this PR, instead of using `data-playwright-test-label` I tried using `getByRole` to illustrate the issue #51770. Of course, if we decide to still use test IDs, I'll update the PR and use them instead.

### CODE REVIEW

File: e2e/signout-modal.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    await page.getByRole('button', { name: translations.buttons.menu }).click();
+    await page
+      .getByRole('button', { name: translations.buttons['sign-out'] })
+      .click();
+
+    await page
+      .getByRole('button', { name: translations.signout.certain })
+      .click();
+
+    await expect(page).toHaveURL('/learn');
```
Comment (naomi-lgbt) [inline e2e/signout-modal.spec.ts:48]: It looks like Webkit visits `/learn/` and the trailing slash is causing this to fail? 🤔

**Review by naomi-lgbt — APPROVED** [review]

File: e2e/signout-modal.spec.ts (hunk 1 above)
Comment (sitek94) [reply e2e/signout-modal.spec.ts]: Hey @naomi-lgbt, thanks! I've fixed it using RegExp, but now for some reason i18n failed 🤔

Comment (naomi-lgbt) [reply e2e/signout-modal.spec.ts]: The failures are unrelated to your PR - I am working on a fix. Thanks for your patience. 💜

Comment (sitek94) [reply e2e/signout-modal.spec.ts]: Alright, great then, thanks!

**Review by Ahmadkashif — APPROVED** [review]
Thanks for pointing out the necessary changes @naomi-lgbt 💯 

This now looks good to go.

**Review by huyenltnguyen — APPROVED** [review]
LGTM 👍 Thanks @sitek94!

**Pull request merged**